File: 2021/day02.ts
import { readFileSync } from "fs";

let part1 = { x: 0, y: 0 };
let part2 = { x: 0, y: 0, aim: 0 };

readFileSync("input.txt", "utf8").trim().split("\n").forEach(l => {
  const [direction, amount] = l.split(" ");
  switch(direction) {
    case "forward":
      part1.x += +amount;
      part2.x += +amount;
      part2.y += (part2.aim * +amount);
      break;
    case "up":
      part1.y -= +amount;
      part2.aim -= +amount;
      break;
    case "down":
      part1.y += +amount;
      part2.aim += +amount;
      break;
  }
});

console.log(part1.x * part1.y);
console.log(part2.x * part2.y);

File: 2021/day02.py
lines = open("input.txt").read().splitlines()

x1, y1 = 0, 0
x2, y2 = 0, 0
aim = 0

for line in lines:
    d, v = line.split(" ")
    v = int(v)
    if d == "forward":
        x1 += v
        x2 += v
        y2 += (aim*v)
    elif d == "down":
        y1 += v
        aim += v
    else:
        y1 -= v
        aim -= v

part1 = x1*y1
part2 = x2*y2

print(part1)
print(part2)
